### ISSUE TITLE
no-loop-func: Functions passed as parameters should not trigger warning

### ISSUE THREAD

**Issue by iancmyers** [eslint/eslint#248]
Functions passed as parameters inside of a loop are erroneously triggering the `no-loop-func` rule, which is meant to catch function declarations inside of a loop. For example, the following will trigger a warning and it shouldn't:

``` js
while (true) {
  foo.reduce(function(a, b) {
    return a + b;
  });
}
```
